src/compiler/checker.ts

```diff
@@ -33152,6 +33187,12 @@ namespace ts {
                     return checkNonNullAssertion(node as NonNullExpression);
                 case SyntaxKind.MetaProperty:
                     return checkMetaProperty(node as MetaProperty);
+                case SyntaxKind.ImportKeyword:
+                case SyntaxKind.NewKeyword:

```

Comment:
It feels a bit odd to put these cases in `checkExpression` for some reason. Is there a reason they need to be here rather than running `checkMetaPropertyExpression` directly in `getSymbolAtLocation`? It feels like nothing should actually have to call `checkExpression` on an `ImportKeyword` or `NewKeyword`; they’re kind of special, and `getSymbolAtLocation`/`getTypeOfNode` handles those kinds of special cases.

Reply:
I was making it act the same way (I think) as, for example, an identifier on the left hand side of a property access expression (it actually looks like it might have been better for me to use `isInExpressionContext` in compiler/utilities.ts for those keywords). I'm not sure which way is desired, but I've updated the code in 635ab3a7f43c36d70961b646cc520776140b8ab0 for us to take a look.

Reply:
I think it’s a gray area, but it seems like we don’t need a case in `checkExpression` because the `import` and `new` keywords in a MetaProperty aren’t even in the parse tree now that you’ve reverted the parser change. I was opting for the minimal possible checker change, and we can always add things in later if we need them.

Reply:
Yeah, sounds good!